openmc/mgxs/library.py

```diff
@@ -239,22 +255,34 @@ def name(self, name):
 
     @mgxs_types.setter
     def mgxs_types(self, mgxs_types):
+        all_mgxs_types = np.append(openmc.mgxs.MGXS_TYPES,
+                                   openmc.mgxs.MDGXS_TYPES)
         if mgxs_types == 'all':
-            self._mgxs_types = openmc.mgxs.MGXS_TYPES
+            self._mgxs_types = all_mgxs_types
         else:
             cv.check_iterable_type('mgxs_types', mgxs_types, basestring)
             for mgxs_type in mgxs_types:
-                cv.check_value('mgxs_type', mgxs_type, openmc.mgxs.MGXS_TYPES)
+                cv.check_value('mgxs_type', mgxs_type, all_mgxs_types)
             self._mgxs_types = mgxs_types
 
     @by_nuclide.setter
     def by_nuclide(self, by_nuclide):
         cv.check_type('by_nuclide', by_nuclide, bool)
+
+        if by_nuclide == True and self.domain_type == 'mesh':
+            raise ValueError('Unable to create MGXS library by nuclide with ' +
```

Comment:
The + is superfluous. String concatenation is implicit when separated only by whitespace.